components/usage/pkg/apiv1/billing.go

```diff
@@ -165,6 +167,76 @@ func (s *BillingService) CreateStripeCustomer(ctx context.Context, req *v1.Creat
 	}, nil
 }
 
+func (s *BillingService) CreateStripeSubscription(ctx context.Context, req *v1.CreateStripeSubscriptionRequest) (*v1.CreateStripeSubscriptionResponse, error) {
+	attributionID, err := db.ParseAttributionID(req.GetAttributionId())
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Invalid attribution ID %s", attributionID)
+	}
+
+	customer, err := s.GetStripeCustomer(ctx, &v1.GetStripeCustomerRequest{Identifier: &v1.GetStripeCustomerRequest_AttributionId{AttributionId: string(attributionID)}})
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, "Stripe customer with attribution ID %s not found", attributionID)
```

Comment:
When this returns an error, you don't know if this is due to a NotFound, or some other error. It could be an internal error as well.

Because you're invoking the RPC directly, you assume it will return the appropriate error (with the right status code) so  you can return that directly.

```suggestion
		return nil, err
```